selfdrive/car/gm/values.py

```diff
@@ -174,6 +176,19 @@ class CanBus:
+  },
+  {
+    170: 8, 190: 6, 192: 5, 193: 8, 197: 8, 199: 4, 201: 6, 208: 8, 209: 7, 211: 2, 241: 6, 249: 8, 288: 5, 289: 1, 290: 1, 298: 8, 300: 1, 304: 8, 309: 8, 311: 8, 313: 8, 320: 8, 322: 7, 328: 1, 352: 7, 368: 8, 369: 8, 381: 6, 386: 5, 388: 8, 393: 7, 398: 8, 407: 4, 413: 8, 417: 7, 419: 1, 422: 4, 426: 7, 431: 8, 442: 8, 451: 8, 452: 8, 453: 6, 454: 8, 455: 7, 460: 5, 462: 4, 463: 3, 479: 3, 481: 7, 485: 8, 487: 8, 489: 8, 497: 8, 499: 3, 500: 6, 501: 8, 508: 8, 510: 8, 512: 3, 530: 8, 532: 6, 534: 2, 554: 3, 560: 8, 562: 8, 563: 5, 564: 5, 573: 1, 608: 8, 609: 6, 610: 6, 611: 6, 612: 8, 613: 8, 647: 6, 707: 8, 761: 7, 801: 8, 803: 8, 804: 3, 810: 8, 821: 4, 823: 7, 832: 8, 840: 5, 842: 6, 844: 8, 866: 2, 961: 8, 967: 4, 969: 8, 977: 8, 979: 8, 985: 5, 1001: 8, 1005: 6, 1009: 8, 1017: 8, 1019: 2, 1020: 8, 1105: 6, 1217: 8, 1221: 5, 1223: 2, 1225: 7, 1233: 8, 1249: 8, 1257: 6, 1265: 8, 1267: 1, 1280: 4, 1300: 8, 1322: 6, 1323: 4, 1328: 4, 1417: 8, 1906: 7, 1907: 7, 1912: 7, 1914: 7, 1917: 7, 1918: 7, 1919: 7, 1920: 7
+  },
+  {
+    170: 8, 190: 6, 192: 5, 193: 8, 197: 8, 199: 4, 201: 8, 208: 8, 209: 7, 211: 2, 241: 6, 249: 8, 288: 5, 289: 1, 290: 1, 298: 8, 300: 1, 304: 8, 309: 8, 311: 8, 313: 8, 320: 8, 322: 7, 328: 1, 352: 7, 368: 8, 369: 8, 381: 5, 386: 5, 388: 8, 393: 7, 398: 8, 407: 4, 413: 8, 417: 7, 419: 1, 422: 4, 426: 7, 431: 8, 442: 8, 451: 8, 452: 8, 453: 6, 454: 8, 455: 7, 460: 5, 462: 4, 463: 3, 479: 3, 481: 7, 485: 8, 487: 8, 489: 5, 497: 8, 499: 3, 500: 6, 501: 8, 508: 8, 510: 8, 512: 3, 530: 8, 532: 6, 534: 2, 554: 3, 560: 8, 562: 8, 563: 5, 564: 5, 573: 1, 608: 8, 609: 6, 610: 6, 611: 6, 612: 8, 613: 8, 647: 6, 707: 8, 761: 7, 801: 8, 803: 8, 804: 3, 810: 8, 821: 4, 823: 7, 832: 8, 840: 5, 842: 6, 844: 8, 866: 4, 961: 8, 967: 4, 969: 8, 977: 8, 979: 8, 985: 5, 1001: 8, 1005: 6, 1009: 8, 1017: 8, 1019: 2, 1020: 8, 1105: 6, 1217: 8, 1221: 5, 1223: 2, 1225: 7, 1233: 8, 1249: 8, 1257: 6, 1265: 8, 1267: 1, 1280: 4, 1300: 8, 1322: 6, 1323: 4, 1328: 4, 1417: 8, 1906: 7, 1907: 7, 1912: 7, 1914: 7, 1917: 7, 1918: 7, 1919: 7, 1920: 7
+  }],
```

Comment:
I'm also unsure if I should have this many fingerprints. I did try merging all of them together into a single dict, but each one had one or more overlapping keys with different values.

Reply:
There was also an issue fingerprinting the 2016 Escalade ESV and when the port was merged, it caused other GM vehicles to fingerprint as that vehicle.

Reply:
Are these all from different runs? You should only need to run the CAN fingerprinting script once, and copy the full dictionary.

As a reminder, it's best to start the script before the car even turns on, then start the car and let it run for a few minutes. Then turn the car off, back on, put it in drive, park, etc. The dict at that point should be pretty complete

Reply:
Did you get a new FP?

Reply:
Those were all from a single run, but it's also possible I didn't let the car run long enough. 

I'll do another run tomorrow and let it go for 5-10 minutes and hopefully get a more complete fingerprint.

Reply:
These are multiple dicts, you should only get one from the script. How did you run it?

Reply:
I ran it via the fingerprinting 1.0 instructions [here](https://github.com/commaai/openpilot/wiki/Fingerprinting#fingerprinting-10).

When I first ran it a few weeks ago, I copied only the last message, but that resulted in a car unrecognized error. When I ran it a second time, I noticed some of the keys were getting different values every few seconds and unless I included the multiple variants, it gave me the car unrecognized error. 

Same thing when I ran it a few days ago. Let it run for 2-4 minutes continuously, then turned it on/off a few times, switched gears, and let it run another minute. I then went back into the output and pulled the fingerprints that are currently in this PR. 

With all of those fingerprints present, it recognizes the car just fine. If I remove any of them (I tried), it fails to recognize the car.

Also fwiw, these fingerprints were all generated with the car stock per the instructions, but the same behavior seems to happen when I have my ASCM bypassed.

Reply:
I think you're just a little confused how the fingerprinting works. The can fingerprinting script reads the can traffic on the bus and records every message address and length it sees. Then during fingerprinting time, it will eliminate cars based on the addresses present on startup. If the car has an address not in a database fingerprint, it will eliminated that car.

So basically, you just need to just run the script with the car off, then turn it on and wait a few minutes. Any address it misses means it may fail to fingerprint later, so leaving it running for longer is best.

Then you only should be left with one dict once you're done (lengths shouldn't change, but you may have more or less addresses based on how long you let it run).

It's possible it's failing to fingerprint because your CAN fingerprint is identical to the Escalade. You can send me a route where it fails (with just one dict) and I can check.

Reply:
My experience with the fingerprinting so far was that it wouldn’t work with all 150 addresses that I gathered.  I saw that one of them did not appear until a few seconds after the car started, maybe because it has to boot up or something. So my theory is that OP was checking the FP before the last one popped onto the network. So I’ve been using the 149 and it works fine even if the car has already started and then OP boots. If I add the 150th one in it always fails. Hopefully my car doesn’t share those 149 with anything else. So anyway I would watch the log when starting the car and see which address may be popping up late, remove that from the fingerprint, and see if it works. Keep in mind that I’m a complete noob here so take my advice at your own risk :-)

Reply:
You want the opposite. You want to include as many addresses as possible, so that when you fingerprint that address doesn't eliminate your car.

Reply:
I updated the fingerprint from a route. Can you try to drive on the branch and make sure it fingerprints properly?

Reply:
Ah I see the issue. One address is on bus 0 and 2, but the length is different! Let me know when you get a drive on the latest commits I made

Reply:
Sorry for the delay and thanks for all your help! Just pulled down your latest commit and started the car, but it didn't recognize it. Here is the route if that's helpful: `ef8f2185104d862e|2023-02-14--15-27-29--0`

Reply:
Okay try that, make sure to pull

Reply:
Boom, that did it. Do you need updated routes?